cci_cameroon/analysis/predictive_model/model_run_report.py

```diff
@@ -0,0 +1,186 @@
+from sklearn.ensemble import RandomForestClassifier
+from sklearn.neighbors import KNeighborsClassifier
+from sklearn.naive_bayes import GaussianNB
+from sklearn.multioutput import MultiOutputClassifier
+from sklearn.svm import SVC
+from nltk.corpus import stopwords
+
+# Project modules
+import cci_cameroon
+from cci_cameroon.pipeline import process_workshop_data as pwd
+from cci_cameroon.pipeline import model_tuning_report as mtr
+
+# %%
+# Set directory
+project_directory = cci_cameroon.PROJECT_DIR
+
+# %%
+os.environ["TOKENIZERS_PARALLELISM"] = "false"
+
+# %%
+stop = stopwords.words("french")
+
+# %%
+# Read train/test data
+X_train = pd.read_excel(
+    f"{project_directory}/outputs/data/data_for_modelling/X_train.xlsx", index_col="id"
+)["comment"]
+X_test = pd.read_excel(
+    f"{project_directory}/outputs/data/data_for_modelling/X_test.xlsx", index_col="id"
+)["comment"]
+y_train = pd.read_excel(
+    f"{project_directory}/outputs/data/data_for_modelling/y_train.xlsx", index_col="id"
+)["category_id"]
+y_test = pd.read_excel(
+    f"{project_directory}/outputs/data/data_for_modelling/y_test.xlsx", index_col="id"
+)["category_id"]
+
+# %%
+# Data to use to train 'no response'
+no_response_train = pd.read_excel(
+    f"{project_directory}/outputs/data/data_for_modelling/no_response_train.xlsx",
+    index_col="id",
+)["comment"]
+
+no_response_test = pd.read_excel(
+    f"{project_directory}/outputs/data/data_for_modelling/no_response_test.xlsx",
+    index_col="id",
+)["comment"]
+
+# %%
+# No reponse - all zeros
+y_nr_train = np.array([[0] * 8] * 120)
+y_nr_test = np.array([[0] * 8] * 30)
+
+# %%
+y_train = y_train.apply(literal_eval)
+y_test = y_test.apply(literal_eval)
+
+# %%
+# Transform Y into multilabel format
+mlb = MultiLabelBinarizer()
+y_train = mlb.fit_transform(y_train)
+y_test = mlb.transform(y_test)
+codes = list(mlb.classes_)  # Codes list
+
+# %%
+y_train = np.concatenate((y_train, y_nr_train))
+y_test = np.concatenate((y_test, y_nr_test))
+
+# %%
+X_train = X_train.append(no_response_train, ignore_index=False)
+X_test = X_test.append(no_response_test, ignore_index=False)
+
+# %%
+X_train, y_train = shuffle(X_train, y_train, random_state=1)
+X_test, y_test = shuffle(X_test, y_test, random_state=1)
+
+# %%
+# Get language model
+model_fr = SentenceTransformer("Sahajtomar/french_semantic")  # French language model
+# Encode train and test transform into word embeddings
+X_train_embeddings_fr = model_fr.encode(list(X_train))
+X_test_embeddings_fr = model_fr.encode(list(X_test))
+
+# %% [markdown]
+# ### Fit and predict with best performing models
+
+# %%
+# Best performing models
+svm = MultiOutputClassifier(SVC(C=3, gamma="scale"), n_jobs=-1)
+knn = KNeighborsClassifier(n_neighbors=5, p=1, weights="distance")
+rf = RandomForestClassifier(n_estimators=200, random_state=1)
+dt = DecisionTreeClassifier(criterion="entropy", random_state=1)
+nb = MultiOutputClassifier(GaussianNB(), n_jobs=-1)
```

Comment:
I think the best model from the tuning script should be used here with the found parameters? So only one model should appear here? Except there is more than one whose performance are at the top